src/API/Management/Users.php

```diff
@@ -1,88 +1,262 @@
+     *      - "connection" name field is required and limited to sms, email, & DB connections.
+     *      - "phone_number" field is required by sms connections.
+     *      - "email" field is required by email and DB connections.
+     *      - "password" field is required by DB connections.
+     *
+     * @return mixed|string
+     *
+     * @throws \Exception
+     *
+     * @link https://auth0.com/docs/api/management/v2#!/Users/post_users
+     */
+    public function create($data)
+    {
+        if (empty($data['connection'])) {
+            throw new \Exception('Missing required "connection" field.');
+        }
 
-        return $this->apiClient->post()
-            ->users()
-            ->withHeader(new ContentType('application/json'))
+        if ('sms' === $data['connection'] && empty($data['phone_number'])) {
+            // "phone_number" field is required for an sms connection.
+            throw new \Exception('Missing required "phone_number" field for sms connection.');
+        } elseif ('sms' !== $data['connection']) {
+            // "email" field is required for email and DB connections.
+            if (empty($data['email'])) {
+                throw new \Exception('Missing required "email" field.');
+            }
+
+            // Passwords are required for DB connections.
+            if ('email' !== $data['connection'] && empty($data['password'])) {
+                throw new \Exception('Missing required "password" field for DB connection.');
+            }
+        }
+
+        return $this->apiClient->method('post')
+            ->addPath('users')
             ->withBody(json_encode($data))
             ->call();
     }
 
-    public function getAll($params = array()) {
-        return $this->search($params);
-    }
-
-    public function search($params = array()) {
+    /**
+     * Search all Users.
+     *
+     * @param array $params - Search parameters to send:
+     *      - "fields", "include_fields", "page", and "per_page" keys here will override the explicit parameters.
+     *      - Queries using "search_engine" set to "v2" should be migrated to v3; see migration @link below
+     * @param null|string|array $fields - Fields to include or exclude from the result, empty to retrieve all fields:
+     *      - Including only the fields required can speed up API calls significantly.
+     *      - Arrays will be converted to comma-separated strings
+     * @param null|boolean $include_fields - True to include $fields, false to exclude $fields.
+     * @param null|integer $page - Page number to get, zero-based.
+     * @param null|integer $per_page - Number of results to get, null to return the default number.
+     *
+     * @return mixed
+     *
+     * @throws \Exception
+     *
+     * @link https://auth0.com/docs/users/search/v3#migrate-from-search-engine-v2-to-v3
+     * @link https://auth0.com/docs/api/management/v2#!/Users/get_users
+     * @link https://auth0.com/docs/users/search/best-practices
```

Comment:
First link is an anchor, the rest are direct. I wouldn't imagine that the migration guide link will change as it's in a number of places.

Reply:
I'd just point to https://auth0.com/docs/users/search/v3 since that's the main page for the search API. Also, in that article there's a link to v2 syntax. So just keep:

- https://auth0.com/docs/api/management/v2#!/Users/get_users (API)
- https://auth0.com/docs/users/search/v3 (User's Search)